microscope/monitor/parser.py

```diff
@@ -78,67 +81,231 @@ def pop_current(self, init: str="") -> str:
+        """
+        get_eps_repr returns tuple with source endpoint
+        and destination endpoint representation
+        """
+        src_repr = ""
+        dst_repr = ""
+        src_ip = ""
+        dst_ip = ""
+        src_port = ""
+        dst_port = ""
+        src_ip_l4 = ""
+        dst_ip_l4 = ""
+
+        try:
+            src_ip, dst_ip = self.get_ips4(event)
+        except (KeyError, StopIteration):
+            pass
+        try:
+            src_ip, dst_ip = self.get_ips6(event)
+        except KeyError:
+            pass
+
+        try:
+            src_port, dst_port = self.get_ports(event)
+        except (KeyError, StopIteration):
+            pass
+
+        if src_port and src_ip:
+            src_ip_l4 = src_ip + ":" + src_port
+
+        if dst_port and dst_ip:
+            dst_ip_l4 = dst_ip + ":" + dst_port
+
+        try:
+            if src_ip:
+                src_ep = self.get_ep_by_ip(src_ip)
+                src_repr = (
+                    f"{src_ep['namespace']}:{src_ep['name']}"
+                )
+        except (KeyError, StopIteration):
+            pass
+
+        try:
+            if not src_repr:
+                src_ep = self.endpoints[event["source"]]
+                src_repr = (
+                    f"{src_ep['namespace']}:{src_ep['name']}"
+                )
+        except KeyError:
+            pass
+        try:
+            if not src_repr:
+                src_repr = self.parse_labels(
+                    self.identities[event["srcLabel"]])
+        except KeyError:
+            if not src_repr:
+                src_repr = str(event["source"])
+
+        if src_ip_l4:
+            src_repr += f" {src_ip_l4}"
+        elif src_ip:
+            src_repr += f" {src_ip}"
+
+        try:
+            if dst_ip:
+                dst_ep = self.get_ep_by_ip(dst_ip)
+                dst_repr = (
+                    f"{dst_ep['namespace']}:{dst_ep['name']}"
+                )
+        except (KeyError, StopIteration):
+            pass
+
+        try:
+            if not dst_repr:
+                dst_ep = self.endpoints[event["dstID"]]
+                dst_repr = (
+                    f"{dst_ep['namespace']}:{dst_ep['name']}"
+                )
+        except KeyError:
+            pass
+        try:
+            if not dst_repr:
+                dst_repr = self.parse_labels(
+                    self.identities[event["dstLabel"]])
+        except KeyError:
+            if not dst_repr:
+                dst_repr = str(event["dstID"])
+
+        if dst_ip_l4:
+            dst_repr += f" {dst_ip_l4}"
+        elif dst_ip:
+            dst_repr += f" {dst_ip}"
+
+        return (src_repr, dst_repr)
+
+    def get_ips4(self, event: Dict) -> Tuple[str, str]:
+        ipv4 = event["summary"]["ipv4"]
+        fields = ipv4.split(" ")
+        src = next(x for x in fields if "SrcIP=" in x).split("=")[1]
+        dst = next(x for x in fields if "DstIP=" in x).split("=")[1]
```

Comment:
Couldn't we json.loads the data instead of this string manipulation? I did notice that the test for these functions uses what looks like a go %#v output which is a surprise. I thought we had switched all output to json (or nested json, but that could be parsed at least)

Reply:
This output comes from `pkg/monitor/dissect.go` which uses Go `gopacket` library to decode all layers. I didn't come up with a good way to decode this information into json, so I decided to parse it in microscope, but it's also doable in the monitor cli.

Reply:
:/ that is really awkward to use. Deep below all the interface I did find a struct (in one case, I didn't try too hard)... but it only had private fields, which means any json dump would have to be manual.

How did we do this before? I guess we were parsing the human-readable equivalent?

If it's all this complex, we could merge this. I'm worried about merging it today, then having to support this particular format from monitor for a long time (since the only toggle we have here is --json being passed to monitor, not a --jsonv1 or something)